contracts/smart-account/modules/AccountRecoveryModule.sol

```diff
@@ -174,22 +184,24 @@ contract AccountRecoveryModule is
                     2) * 65];
 
                 currentGuardianAddress = keccak256(
-                    abi.encodePacked(CONTROL_MESSAGE, userOp.sender)
+                    abi.encodePacked(CONTROL_MESSAGE, smartAccount)
                 ).toEthSignedMessageHash().recover(currentGuardianSig);
 
                 if (currentUserOpSignerAddress != currentGuardianAddress) {
                     return SIG_VALIDATION_FAILED;
                 }
             }
 
-            address sender = userOp.sender;
+            //address sender = userOp.sender;
             bytes32 currentGuardian = keccak256(currentGuardianSig);
 
-            uint48 validAfter = _guardians[currentGuardian][sender].validAfter;
-            uint48 validUntil = _guardians[currentGuardian][sender].validUntil;
+            uint48 validAfter = _guardians[currentGuardian][smartAccount]
+                .validAfter;
+            uint48 validUntil = _guardians[currentGuardian][smartAccount]
+                .validUntil;
```

Comment:
Fetching both `validAfter` and `validUntil` in a single call can reduce gas.
```solidity
TimeFrame memory guardianTimeFrame = _guardians[currentGuardian][smartAccount];

uint48 validAfter = guardianTimeFrame.validAfter;
uint48 validUntil = guardianTimeFrame.validUntil;
```

Please double check

Reply:
>Fetching both validAfter and validUntil in a single call can reduce gas.

like this?

```
(uint48 validAfter, uint48 validUntil) = (
                _guardians[currentGuardian][smartAccount].validAfter, 
                _guardians[currentGuardian][smartAccount].validUntil
            );
```

Reply:
Not sure, the goal is to access to storage once here
that's why the `TimeFrame memory guardianTimeFrame = _guardians[currentGuardian][smartAccount];`

Reply:
got it. yep, it saves about 150 gas. fixed, thnx!